lib/pbench/agent/tool_meister.py

```diff
@@ -1534,37 +1540,27 @@ def end_tools(self, data):
                 )
                 failures += 1
 
-        # Remove persistent tool temporary working directory on remote host
+        # Remove persistent tool temporary working directory
         directory = data["directory"]
         tool_dir = self.directories[directory]
 
-        if self._hostname == self._controller:
-            directory_to_delete = tool_dir
-        else:
-            directory_to_delete = tool_dir.parent
-
         self.logger.debug(
             "%s: deleting persistent tool tmp directory %s %s",
             self._hostname,
             self._group,
-            directory_to_delete,
+            tool_dir,
         )
-        unexpected_files = [x for x in directory_to_delete.iterdir() if x.is_file()]
-        if not unexpected_files:
-            try:
-                shutil.rmtree(directory_to_delete)
-            except Exception:
-                self.logger.exception(
-                    f"Failed to remove tmp persistent tool data hierarchy, "
-                    f"{directory_to_delete}"
+        for dirpath, _, files in os.walk(tool_dir):
+            if files:
+                self.logger.warning(
+                    f"Found regular files {files} at {dirpath} and will be deleted later"
```

Comment:
"found [this] and will be deleted later" isn't grammatical. I also don't find the reference to "regular files" or the "will be deleted later" to be of much real value in what could turn out to be a very long message. (Also, "later" is really "immediately following this message", which means the distinction isn't very useful.)

Including the full directory path only helps in that it includes a hostname, which would otherwise be ambiguous; but I'd kinda prefer that the remote generating the message identify itself explicitly (and more compactly) instead. Listing the files, if we're implicitly deleting them, probably makes sense, but let's drop at least some of the boilerplate around the list, and put the list last.

Perhaps something like `"{hostname} unexpected temp files [{','.join(files)}]`???

Also, I think showing a separate `warning` message, from the same place, for each directory in the tree, is confusing. (Granted, we don't expect this to happen a lot.) It's only a little more work to aggregate a list that's dumped in a single message. (For bonus points, produce relative paths from the root temp directory with `Path.relative_to(*other)`.)

Reply:
Yes I knew you would have something better to offer in terms of messaging here, so I didnt give a lot of thought to format this message :stuck_out_tongue_closed_eyes:

> Perhaps something like `"{hostname} unexpected temp files [{','.join(files)}]`???
> 
> Also, I think showing a separate `warning` message, from the same place, for each directory in the tree, is confusing. (Granted, we don't expect this to happen a lot.) It's only a little more work to aggregate a list that's dumped in a single message. (For bonus points, produce relative paths from the root temp directory with `Path.relative_to(*other)`.)

Yes I concur.